_posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md

```diff
@@ -0,0 +1,163 @@
+
+And here's the result: [my site on GitHub](https://github.com/ThatRobVK/ThatRobVK.github.io). Ready to rock!
+
+I've noticed a really neat feature in GitHub. It has automatically created an environment and set up a deployment to GitHub Pages for me. Must be because I've named the repository in the correct way - clever! The site currently displays a 404, but it's a good start.
+
+![Environment automatically set up in GitHub](/images/2019-04-15-Environment.png)
+
+#### Multi-domain setup on GitHub Pages
+
+DuckDuckGo and Stack Overflow to the rescue! First hit, it seems you can create a CNAME file in a folder with the domain name in it. When you then point your DNS at GitHub, it'll route to the correct repository. Let's give it a go.
+
+According to [the GitHub docs](https://help.github.com/en/articles/setting-up-an-apex-domain#configuring-an-alias-or-aname-record-with-your-dns-provider) you can set up an `ALIAS` record. My domain is registered through AWS Route53 and I happen to know they support `ALIAS` records, so let's try that! I create an alias to `thatrobvk.github.io` - nope, AWS doesn't like that. Looks like [you can only create alias records to AWS resources](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html). Funnily enough that was what I thought, but if you don't try you don't learn, right?
+
+OK, good old `A` and `CNAME` records it is. First the root domain name. [GitHub provides instructions here](https://help.github.com/en/articles/setting-up-an-apex-domain#configuring-a-records-with-your-dns-provider) and that's done in minutes. Now for the 1-2 day wait for DNS propagation...
+
+Meanwhile, I notice a sentence `before configuring your custom domain with your DNS provider, add your custom domain to your GitHub Pages site` - Whoops! Let's quickly do that. There's a [GitHub quick start guide](https://help.github.com/en/articles/quick-start-setting-up-a-custom-domain) that shows you how to do it.
+
+![GitHub custom domain configuration](/images/2019-04-15-CustomDomain.png)
+
+You can only add a single domain through the UI. So what about `www` and my blog subdomain? Let's see what it's done to the files. When I do a `git pull` I get a `CNAME` file down with `robvk.uk` as its contents.
+
+At this point I spent a little time digging into whether I could get the DNS for the subdomain to point to a subfolder on the primary domain. I'm pretty sure DNS won't do this and normally you'd configure the web server to route traffic by domain to a website on the server. I did read an amusing thread on a forum with one person becoming quite irate whilst repeatedly claiming their work DNS did exactly this. I gave it a go anyway, just in case. Even though Route53 accepts a full URL as the target for a `CNAME`, it doesn't work.
+
+{: .um-actually}
```

Comment:
does this work on the IW blog?

Reply:
Do you mean the {: .um-actually}? No it doesn't, but ends up putting in a class that doesn't exist. I'll update it to remove that line, it's cleaner.

Reply:
I have removed the tag